test/com/google/javascript/jscomp/CollapsePropertiesTest.java

```diff
@@ -202,11 +274,22 @@ public void testGlobalObjectNameInBooleanExpressionDepth3() {
   public void testGlobalFunctionNameInBooleanExpressionDepth1() {
     test("function a() {} a.c = 1; if (a) x(a.c);",
          "function a() {} var a$c = 1; if (a) x(a$c);");
+
+    test("function a() {} /** @nocollapse */ a.c = 1; if (a) x(a.c);",
+        "function a() {} a.c = 1; if (a) x(a.c);");
   }
 
   public void testGlobalFunctionNameInBooleanExpressionDepth2() {
     test("var a = {b: function(){}}; a.b.c = 1; if (a.b) x(a.b.c);",
          "var a$b = function(){}; var a$b$c = 1; if (a$b) x(a$b$c);");
+
+    test("var a = {/** @nocollapse */ b: function(){}}; a.b.c = 1; "
+        + "if (a.b) x(a.b.c);",
+        "var a = {b: function(){}}; var a$b$c = 1; if (a.b) x(a$b$c);");
```

Comment:
I specifically implemented this such that `@nocollapse` should not impact sub-properties. In this case, I believe `a.b.c`  should be fully collapsed and not remain a property since it doesn't have its own `@nocollapse` annotation.